client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx

```diff
@@ -94,15 +94,7 @@ const DomainsStep: Step = function DomainsStep( { navigation, flow } ) {
 			} );
 			dispatch( submitDomainStepSelection( suggestion, getAnalyticsSection() ) );
 
-			let hideFreePlan;
-
-			if ( flow === DOMAIN_UPSELL_FLOW ) {
-				hideFreePlan = true;
-			} else {
-				hideFreePlan = Boolean( suggestion.product_slug ) || shouldHideFreePlan;
-			}
-
-			setHideFreePlan( hideFreePlan );
+			setHideFreePlan( Boolean( suggestion.product_slug ) || shouldHideFreePlan );
```

Comment:
Just want to clarify what's happening here. Looks like we're taking out special logic for the DOMAIN_UPSELL_FLOW. I think we can do that here - without affecting other flows, because we're setting we're invoking `setHideFreePlan( true );` on line 51 in [domain-upsell.ts](https://github.com/Automattic/wp-calypso/pull/74897/files#diff-528d0329c8ad2a1886e8ceeacf51f5b4f639a264bae1ea37826c1b5080846a43). 

So basically, we just update that value in state when working within the domain-upsell flow, and the rest of the logic just works down stream. 

Assuming that's right, I wonder whether there would be any unanticipated effects of updating that state value? For example, once we invoke `setHideFreePlan( true );` is it possible that a user would end up back on the domains page via some other route/flow where they SHOULD see the free plan, but not see it because we've updated that state value?

Reply:
We're moving the "hide free plan" logic into the domain-upsell flow itself, instead of having that logic inside the domains step. It's essentially the same thing, but this change allows us to ensure that the Free plan is always hidden for the domain-upsell flow without modifying other code in the plans or domain step (until we decide we want to show the free plan for specific scenarios within the domain-upsell flow).

> Assuming that's right, I wonder whether there would be any unanticipated effects of updating that state value? For example, once we invoke setHideFreePlan( true ); that means as long as that state value persist, we'd never show the free plans. Is it possible that a user would end up back on the domains page via some other route/flow where they SHOULD see the free plan, but not see it because we've updated that state value?

This is definitely possible if the same redux state value is retrieved/read in another domain-related component, but that will happen regardless of these changes. The only reference for `getHideFreePlan` (to retrieve the hide free plan value from redux) in stepper `plans/plans-wrapper.tsx`. This plans component is only used in Link-in-bio, newsletter, and the domain-upsell flow. The link-in-bio and newsletter will not be affected by these changes since they use the "old" plans component design (vs the new plans grid design we see on the /plans screen when going the domain-upsell flow)